src/lib.rs

```diff
@@ -183,12 +181,7 @@ cfg_if! {
         #[path = "windows_uwp.rs"] mod imp;
     } else if #[cfg(windows)] {
         #[path = "windows.rs"] mod imp;
-    } else if #[cfg(all(target_arch = "x86_64", any(
-                  target_os = "hermit",
-                  target_os = "l4re",
-                  target_os = "uefi",
-                  target_env = "sgx",
-              )))] {
+    } else if #[cfg(all(target_arch = "x86_64", target_env = "sgx"))] {
         #[path = "rdrand.rs"] mod imp;
     } else if #[cfg(feature = "custom")] {
         use custom as imp;
```

Comment:
In the OP you wrote the following:

>This crate also directly exposes a cpurand function for cases where a user want CPU-based RNG, even on platforms that have other options (like Linux)

But with the current configuration custom RNG will never be used for targets with a supported OS. To make it work as intended we may allow custom RNGs for all targets (i.e. move `if #[cfg(feature = "custom")]` to the very beginning), but it will have certain security implications, since users will be able to overwrite randomness source with a potentially insecure one even on well supported targets like Linux and Windows. I don't think it's a big deal and it even may be useful for niche use cases like deterministically testing code which uses `getrandom`, but it should be an explicit choice.

Reply:
So I rephrased the OP to be more clear. The `getrandom-cpu` crate adds a custom `getrandom` handler that only runs on supported architectures. It also directly exposes a function `getrandom_cpu::cpurand` that allows a user of the crate to directly access the CPU randomness. 

This made it possible to test the RDRAND implementation on Linux. It also makes sure that supported targets cannot be overridden. This seems like the best of both worlds. Does this seem reasonable?

Reply:
I am not sure about usefulness of `getrandom_cpu::cpurand` in the proposed form. At the very least instead of independent function it could be better to define a struct which will implement `RngCore + CryptoRng`, so it will be possible to use it as part of the `rand` stack on "supported" platforms.

Reply:
I think you're correct here, we don't need to export another function. With the changes to use a `cpu` Cargo feature, exporting any `cpurand` function is no longer needed.